Server/VikingVault.Services/UnqiueEmailService.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using VikingVault.DataAccess;
+using VikingVault.Services.Abstractions;
+
+namespace VikingVault.Services
+{
+    public class UniqueEmailService : IUniqueEmailService
+    {
+        private readonly VikingVaultDbContext _dbContext;
+
+        public UniqueEmailService(VikingVaultDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+
+        public bool? IsUniqueEmail(string email)
+        {
+
+            try
+            {
+                var ReturnedEmail = _dbContext.User.SingleOrDefault(s => s.Email == email);
+
+                if (ReturnedEmail == null) // No email found
+                    return true;
+
+                else // found it
+                    return false;
+            }
+            catch(Exception e)
+            {
+                return null;
```

Comment:
This approach is less than optimal. We should propagate the exception or send a Wrapped Exception instead. Returning null can be confusing since we don`t have any context on the upper levels regarding what went wrong.